### PR TITLE
Homework lesson 13 class inheritance

### PR DESCRIPTION
**Pull request by sorinmosut**

### CODE REVIEW

**Review by vlad-o-git — COMMENTED** [review]
From what I understood and based on your code, it seems the PerimeterMixin is a base class of Polygons. So Polygons is a child of PerimeterMixin which I don't think this was Marius plan. The __mro__ should display the mixin class between Triangle and Polygons

**Comment by sorinmosut** [comment]
I done it like this becouse we will can use the perimeter method for any geometric figure defined by a class. The only restriction in calling the mixin class is that those sould be call first in this case like this :  Polygons(PerimeterMixin, object), but the object is implicitly. So the __init__ is in Polygons and in mixin is only one method.
[<class '__main__.Polygons'>, <class '__main__.PerimeterMixin'>, <class 'object'>]

File: lesson_13/mosut_sorin/hw_lesson13_inheritance_mosut_sorin.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+from math import sqrt
+
+
+class PerimeterMixin:
+
+    def perimeter(self):
+        return f'The perimeter of a {self.geometric_figure} with sides {self.sides} is {sum(self.sides):.2f}\n'
+
+
+class Polygons(PerimeterMixin):
```
Comment (Liitle) [inline lesson_13/mosut_sorin/hw_lesson13_inheritance_mosut_sorin.py]: The requirement of the challenge was to have the perimeter method available only for triangle class. If we use it like this we will have perimeter method on all the shapes that use Polygons as a parent

File: lesson_13/mosut_sorin/hw_lesson13_inheritance_mosut_sorin.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def __str__(self):
+        printing_format = str()
+        for side_index, length in enumerate(self.sides, start=1):
+            printing_format += 'Side {} with length: {:.2f}\n'.format(side_index, length)
+        return printing_format
+
+    def area(self):
```
Comment (Liitle) [inline lesson_13/mosut_sorin/hw_lesson13_inheritance_mosut_sorin.py:100]: This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create a new object we need the class object (cls).

What we can use from python to achieve this?

**Review by Liitle — CHANGES_REQUESTED** [review]

**Review by Liitle — APPROVED** [review]

**Pull request merged**